index.js

```diff
@@ -39,8 +39,13 @@ function zipFiles(files, filename, source, destination, info, verbose) {
 function pack({ source, destination, info, verbose }) {
     return packlist({ path: source })
         .then(files => {
-            return getDefaultOuputFilename({ cwd: source })
+            const destinationIsDirectory = fs.existsSync(destination) && fs.lstatSync(destination).isDirectory();
+            return getDefaultOuputFilename({ cwd: source , filename: destinationIsDirectory ? undefined : path.basename(destination)})
```

Comment:
Instead of calling the getDefaultName helper always to check if the destination has a filename, can we check here instead and skip the call entirely

Reply:
Although, if we are no longer loading the package.json file to parse the name we may want to add a check to verify that we have a valid package at a minimum. Scratch the above comment and instead update the function to be verifyPackageOutputFileName